pkg/sentry/fsimpl/gofer/filesystem.go

```diff
@@ -678,6 +678,16 @@ func (fs *filesystem) LinkAt(ctx context.Context, rp *vfs.ResolvingPath, vd vfs.
 		if d.nlink == math.MaxUint32 {
 			return linuxerr.EMLINK
 		}
+		if d.isSynthetic() {
+			parent.createSyntheticChildLocked(&createSyntheticOpts{
```

Comment:
eh can you also mention that d.nlink is not updated and represents the incorrect value for synthetic files because hard links are not supported fully in the gofer client yet.

To support it correctly, we need to introduce the inode abstraction; i.e. another struct which contains inode specific informations like timestamps, nlink, etc and each dentry points to an inode. hard linked dentries point to the same inode. One day we will implement it.